File: 09-Backspace-String-Compare/backspaceCompare.py
def process(s: str) -> str:
    result = []
    for i in s:
        if i != "#":
            result.append(i)
        else:
            if result:
                result.pop()
    return "".join(result)


def backspaceCompare(s: str, t: str) -> bool:
    return process(s) == process(t)


if __name__ == "__main__":
    s = "ab#c"
    t = "ad#c"
    print(backspaceCompare(s, t))

File: 09-Backspace-String-Compare/backspaceCompare.go
package main

import (
	"fmt"
	"strings"
)

func process(s string) string {
	result := []string{}
	for _, v := range s {
		if v == '#' {
			if len(result) > 0 {
				result = result[:len(result)-1]
			}
		} else {
			result = append(result, string(v))
		}
	}
	return strings.Join(result, "")
}

func backspaceCompare(s string, t string) bool {
	return process(s) == process(t)
}

func main() {
	result := []string  {"a", "b", "c", "d", "e"}
	result = result[0:2]

	S := "ab#c"
	T := "ad#c"

	fmt.Println(backspaceCompare(S, T))
	
}
